Browser/keywords/interaction.py

```diff
@@ -599,6 +612,15 @@ def handle_future_dialogs(self, action: DialogAction, prompt_input: str = ""):
                 Request().AlertAction(alertAction=action.name, promptInput=prompt_input)
             )
             logger.debug(response.log)
+            if assertion_operator is not None:
+                dialog_text = response.body
+                logger.debug(dialog_text)
+                verify_assertion(
+                    dialog_text,
+                    assertion_operator,
+                    assertion_expected,
+                    f"Asserting dialog text",

```

Comment:
Remove the `f` from the front of the string and CI will go further.

Reply:
Thanks !
That f character was not required anymore.

But I think the reason why I can't get this working is because the dialog text is not passed through.
When you look at the interaction.py code I log 2 things (the log, and the body.... The body should contain the dialog text. But it doesn't. That the real problem.... it isn't returned from the invokeOnPage function.

you can see this also in the robot log:

DEBUG	Test timeout 30 seconds active. 29.605 seconds left.	
DEBUG	Set event handler for next alert	
DEBUG		
DEBUG	Unknown error received: Asserting dialog text '' (str) should be 'Am an alert' (str)

The third line (corresponding with logging of the dialog text) is empty

Reply:
Ah, now I see the problem and remember how it works, sorry I did lead you to wrong direction. It due of the promises we use. 

When you have this test:
```
Dismiss Alert
    Handle Future Dialogs    action=dismiss
    Click    \#alerts
```
The `Handle Future Dialogs` sets a promise in the node side to handle the future alert. But when `Handle Future Dialogs` keyword is run, the alert is not yet there. The alert is only present when `Click` keyword is run and alert is handled in the background by playwright (Handle Future Dialogs).

But I think, means did not test, using promise should work:
```
Dismiss Alert
    ${promise}    Promise To    Handle Future Dialogs    action=dismiss
    Click    \#alerts
    ${text}=    Wait For  ${promise}
```
The point which makes me unsure that does promise and assertion engine work together. It might not work or then again it might work, no idea basically. Gut feeling says that it might not work.

Reply:
My guess is that the problem is not the `invokeOnPage` , the problem is the dialog.text() returning undefined... are you sure the problem is at the invokeOnPage function?

Reply:
If it does not work, then you can return the text in the dialogue and assert the text by RF core keyword. It is not as coinvent as assertion engine, but could be still useful.

Reply:
I don't think promise and the current event handler implementation will play nicely together. Might be necessary to change the ts side impl from a `await invokeOnPage(page, 'on', 'dialog', fn)` to `await invokeOnPage(page, 'waitFor', 'dialog', fn)`. But it might make sense to make that as a separate keyword even.